pkg/compiler/global_test.go

```diff
@@ -55,3 +55,18 @@ func TestMultiDeclarationLocalCompound(t *testing.T) {
 	}`
 	eval(t, src, big.NewInt(6))
 }
+
+func TestMultipleLocalsWithTheSameName(t *testing.T) {
+	src := `package foo
+	func Main() int {
+		a := 1
+		if true {
```

Comment:
But what `if false`? Wouldn't there be a problem of inner `a` affecting the outer one?

Reply:
In this case (with `false`), no. It may in other cases (when assignment to `a` is _executed_ both inside and outside if). We don't support shadowing fully (and never did I believe, as we identify variables only by name).

Reply:
Related e2101523

Reply:
Do we have a bug for it?

Reply:
Of course we have! #1131

Reply:
Updated PR with full support for shadowing.